src/app/stream-deck/StreamDeckSettings/StreamDeckSettings.tsx

```diff
@@ -0,0 +1,73 @@
+import Switch from 'app/dim-ui/Switch';
+import { t } from 'app/i18next-t';
+import { AppIcon, faArrowCircleDown } from 'app/shell/icons';
+import { useThunkDispatch } from 'app/store/thunk-dispatch';
+import {
+  startStreamDeckConnection,
+  stopStreamDeckConnection,
+  streamDeckChangeStatus,
+} from 'app/stream-deck/actions';
+import {
+  generateIdentifier,
+  resetStreamDeckAuthorization,
+} from 'app/stream-deck/authorization/authorization';
+import { streamDeckConnectedSelector, streamDeckEnabledSelector } from 'app/stream-deck/selectors';
+import { streamDeckLocal } from 'app/stream-deck/util/local-storage';
+import { useSelector } from 'react-redux';
+import './StreamDeckSettings.scss';
+
+export const $streamDeckFeature = ['beta', 'dev'].includes($DIM_FLAVOR);
+
+export default function StreamDeckSettings() {
+  const dispatch = useThunkDispatch();
+  const connected = useSelector(streamDeckConnectedSelector);
+  const enabled = useSelector(streamDeckEnabledSelector);
+
+  const onPluginInstall = () => {
+    window.open('https://apps.elgato.com/plugins/com.dim.streamdeck');
+  };
+
+  const onStreamDeckChange = async (enabled: boolean) => {
+    streamDeckLocal.setEnabled(enabled);
+    dispatch(streamDeckChangeStatus(enabled));
+    generateIdentifier();
+    if (enabled) {
+      dispatch(startStreamDeckConnection());
+    } else {
+      dispatch(stopStreamDeckConnection());
+    }
+  };
+
+  const onStreamDeckAuthorizationReset = () => {
```

Comment:
I like to see some light documentation that explains what these are for

Reply:
I moved the core of the feature to 

`async-module.ts`

where I export

```ts
// async module loaded in ./stream-deck.ts using lazy import
export default {
  startStreamDeckConnection,
  stopStreamDeckConnection,
  streamDeckSelectItem,
  streamDeckSelectLoadout,
  resetIdentifierOnStreamDeck,
}
```

that is lazy imported in `stream-deck.ts`, the first one of this functions is invoked

```ts
function lazyLoad(fn: string) {
  return (...args: any[]): ThunkResult =>
    async (dispatch: DimThunkDispatch, getState: () => RootState) => {
      const sd = await loadStreamDeckAsync();
      return sd[fn].call(null, dispatch, getState, ...args);
    };
}

export const startStreamDeckConnection = lazyLoad('startStreamDeckConnection');
export const stopStreamDeckConnection = lazyLoad('stopStreamDeckConnection');

export const streamDeckSelectItem = lazyLoad('streamDeckSelectItem');
export const streamDeckSelectLoadout = lazyLoad('streamDeckSelectLoadout');
```

any thoughts  about this ?

Reply:
Seems like it would work. I think there's also a way to register a new reducer with the redux store that might be simpler overall, but the intent is met.